ompi/mca/mtl/ofi/mtl_ofi_component.c

```diff
@@ -620,6 +620,15 @@ ompi_mtl_ofi_component_init(bool enable_progress_threads,
     hints->caps |= FI_HMEM;
     hints->domain_attr->mr_mode |= FI_MR_HMEM | FI_MR_ALLOCATED;
 
+    /*
+     * Libfabric providers choose whether to perform CUDA memory transfers when
+     * FI_HMEM_CUDA_ENABLE_XFER is unspecified. Some providers may choose to
+     * disable CUDA memory transfers by default. Explicitly enabling this option
+     * allows providers to perform CUDA memory transfers via FI_HMEM if
+     * supported.
+     */
+    setenv("FI_HMEM_CUDA_ENABLE_XFER", "1", 1);
```

Comment:
This flag is currently only required for the OFI BTL when using atomics, and SHOULD BE unset everywhere else to ensure that we use GDR Copy.  By setting this flag for MTL, you are forcing EFA to use CUDA Memcpy (everywhere), when we

Reply:
These are all Libfabric/EFA provider implementation details which should not block the changes proposed in this PR. The only point of this PR is to tell Libfabric it's safe to make CUDA memory transfers, which is true for Open MPI; not true for something like AWS-OCI-NCCL.

Reply:
I agree with you that it is a provider implementation detail.